File: platform-mobile/src/main/java/ru/pocketbyte/locolaser/platform/mobile/resource/IosBaseClassResources.java
package ru.pocketbyte.locolaser.platform.mobile.resource;

import ru.pocketbyte.locolaser.resource.AbsPlatformResources;

import java.io.File;

public abstract class IosBaseClassResources extends AbsPlatformResources {

    private String mTableName;

    public IosBaseClassResources(File resourcesDir, String name, String tableName) {
        super(resourcesDir, name);
        mTableName = tableName;
    }

    public void setTableName(String tableName) {
        mTableName = tableName;
    }

    public String getTableName() {
        if (mTableName != null)
            return mTableName;
        return getDefaultTableName();
    }

    protected String getDefaultTableName() {
        return "Localizable";
    }
}

File: platform-mobile/src/main/java/ru/pocketbyte/locolaser/platform/mobile/resource/IosBaseClassResources.kt
package ru.pocketbyte.locolaser.platform.mobile.resource

import ru.pocketbyte.locolaser.resource.AbsPlatformResources

import java.io.File

abstract class IosBaseClassResources(
        resourcesDir: File,
        name: String,
        tableName: String?,
        filter: ((key: String) -> Boolean)?
) : AbsPlatformResources(resourcesDir, name, filter) {

    var tableName: String? = null
        get() = field ?: defaultTableName

    protected val defaultTableName: String
        get() = "Localizable"

    init {
        this.tableName = tableName
    }
}
